Cal 242 replace documentation with draft-documentation

ricklarsen opened the pull request:
#### What does this PR do?

- Replaces documentation with new outline.
- Removes outdated contents files from old outline.
- Minor cleanup and formatting.

#### Who is reviewing it?

@mcalcote @vinamartin @Lambeaux @jrnorth 

#### Choose 2 committers to review/merge the PR.
@coyotesqrl
@figliold
@jlcsmith
@kcwire
@lessarderic
@pklinef
@shaundmorris
@stustison
#### How should this be tested?

- builds successfully
- no missing sections (search `unresolved directive` in finished pdf/html

#### Any background context you want to provide?

depends on DDF PR: https://github.com/codice/ddf/pull/1615

sections 12-13 are revised in these PRs:
- https://github.com/codice/ddf/pull/1594
- https://github.com/codice/alliance/pull/239

#### What are the relevant tickets?

[CAL-242](https://codice.atlassian.net/browse/CAL-242)

#### Screenshots (if appropriate)

See attached pdf: 
[documentation.pdf](https://github.com/codice/alliance/files/743681/documentation.pdf)


#### Checklist:
- [x] Documentation Updated
	- [ ] Change Log Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

vinamartin commented on distribution/docs/src/main/resources/_contents/_introduction/introduction-contents.adoc:
```diff
@@ -1,11 +1,13 @@
+Codice ${cal-branding} is an open source, modular integration framework building on ${ddf-branding} with the addition of DoD/IC and NATO support capabilities.
 
-=== ${branding} Features
+==== ${cal-branding} Features
```
What does this expand to and what file is this defined in?

ricklarsen replied:
`<cal-branding>Alliance</cal-branding>` from the top-level pom file: 

https://github.com/codice/alliance/blob/master/pom.xml#L83

vinamartin replied:
Just curious, if they both expand to the same thing, why change it here?

ricklarsen replied:
if these docs were to be imported into a custom project, the `${branding}` can change to that project, whereas `${cal-branding}` should be set to resolve to the name of this project.

vinamartin replied:
Ah okay makes sense to me!

ricklarsen commented:
❗️ Looks like ${catalog-ui}, ${ddf-registry} and ${ddf-resource-management} don't get replaced properly in the PDF.

`fixed`
❓ Did we put the Quickstart section in the middle of the document on purpose?

`I'll look back through older commits, but it was suggested on a previous PR to put the installation instructions before the quick start. I liked it at the beginning myself. `

❓ Any reason we repeat Version: 0.2-SNAPSHOT at the beginning of certain sections? Couldn't we just have it showing up once on the cover page? Or maybe in the page header or footer?

`It's a holdover from when applications could theoretically have different version numbers. It can be dropped if it doesn't add anything. `

❓ I noticed that the sections in Appendix B all start with B., but the section numbers in Appendix A are just a continuation of the rest of the document. Shouldn't we be consistent?

`fixed`

❓ Any way we could add the major section headings (e.g., Overview, Core Concepts, Managing, Developing, etc.) to the TOC? I think it would help clarify the structure and navigate the document.

`Yes. It forces a restart of the section numbering to do so though. Looking into a fix for it`

ricklarsen commented:
- Found a way to create major section headers without restarting section numbering. 
- Dropped the superfluous version number references.
- Fixed appendix numbering
- leaving the quickstart where it is for now. I'll try to get some consensus for its placement.
